File: SchoolBack/src/main/kotlin/com/open_inc/SchoolBack/models/Subject.kt
package com.open_inc.SchoolBack.models

import java.time.OffsetDateTime
import jakarta.persistence.*

@Entity
@Table(name = "Subject")
data class Subject(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,

    @Column(name = "name", length = 32, unique = true, nullable = false)
    val name: String,

    @Column(name = "price", precision = 2)
    val price: Float = 0f,

    @Column(name = "created_at", nullable = false)
    val createdAt: OffsetDateTime = OffsetDateTime.now(),

    @Column(name = "updated_at", nullable = false)
    val updatedAt: OffsetDateTime = OffsetDateTime.now()
)

File: SchoolFront/lib/dataclasses/subject.dart
class Subject {
  final int id;
  final String name;
  final double price;
  String? _level;

  String get level => _level??"Уровень не определен";
  void set level(String level){
    _level = level;
  }

  Subject({
    String? level,
    this.id = 0,
    required this.name,
    required this.price
  }):_level=level;

  String toString(){
    return "${name}";
  }
}
